test/k8sT/gsg.go

```diff
@@ -0,0 +1,179 @@
+		kubectl.ValidateNoErrorsOnLogs(CurrentGinkgoTestDescription().Duration)
+		Expect(microscopeCancel()).To(BeNil(), "cannot stop microscope")
+	})
+
+	AfterEach(func() {
+		ExpectAllPodsTerminated(kubectl)
+		for _, manifest := range toDelete {
+			kubectl.Delete(manifest)
+		}
+	})
+
+	It("GRPC example", func() {
+		AppManifest := helpers.GetFilePath("../examples/kubernetes-grpc/cc-door-app.yaml")
+		PolicyManifest := helpers.GetFilePath("../examples/kubernetes-grpc/cc-door-ingress-security.yaml")
+
+		clientPod := "terminal-87"
+
+		toDelete = append(toDelete, AppManifest)
+		toDelete = append(toDelete, PolicyManifest)
+
+		By("Testing the example config")
+		kubectl.Apply(AppManifest).ExpectSuccess("cannot install the GRPC application")
+
+		err := kubectl.WaitforPods(helpers.DefaultNamespace, "-l zgroup=grpcExample", 300)
+		Expect(err).Should(BeNil(), "Pods are not ready after timeout")
+
+		res := kubectl.ExecPodCmd(
+			helpers.DefaultNamespace, clientPod,
+			"python3 /cloudcity/cc_door_client.py GetName 1")
+		res.ExpectSuccess("Client cannot get Name")
+
+		res = kubectl.ExecPodCmd(
+			helpers.DefaultNamespace, clientPod,
+			"python3 /cloudcity/cc_door_client.py GetLocation 1")
+		res.ExpectSuccess("Client cannot get Location")
+
+		res = kubectl.ExecPodCmd(
+			helpers.DefaultNamespace, clientPod,
+			"python3 /cloudcity/cc_door_client.py SetAccessCode 1 999")
+		res.ExpectSuccess("Client cannot set Accesscode")
+
+		By("Testing with L7 policy")
+		_, err = kubectl.CiliumPolicyAction(
+			helpers.DefaultNamespace, PolicyManifest,
+			helpers.KubectlApply, 300)
+		Expect(err).To(BeNil(), "Cannot import GPRC policy")
+
+		res = kubectl.ExecPodCmd(
+			helpers.DefaultNamespace, clientPod,
+			"python3 /cloudcity/cc_door_client.py GetName 1")
+		res.ExpectSuccess("Client cannot get Name")
+
+		res = kubectl.ExecPodCmd(
+			helpers.DefaultNamespace, clientPod,
+			"python3 /cloudcity/cc_door_client.py GetLocation 1")
+		res.ExpectSuccess("Client cannot get Location")
+
+		res = kubectl.ExecPodCmd(
+			helpers.DefaultNamespace, clientPod,
+			"python3 /cloudcity/cc_door_client.py SetAccessCode 1 999")
+		res.ExpectFail("Client can set Accesscode and it shoud not")
+	})
+
+	It("Minikube intro", func() {
+		AppManifest := helpers.GetFilePath("../examples/minikube/http-sw-app.yaml")
+		L3Policy := helpers.GetFilePath("../examples/minikube/sw_l3_l4_policy.yaml")
+		L7Policy := helpers.GetFilePath("../examples/minikube/sw_l3_l4_l7_policy.yaml")
+
+		toDelete = append(toDelete, AppManifest)
+		toDelete = append(toDelete, L3Policy)
+		toDelete = append(toDelete, L7Policy)
+
+		kubectl.Apply(AppManifest).ExpectSuccess("Cannot apply example manifest")
+
+		err := kubectl.WaitforPods(helpers.DefaultNamespace, "", 300)
+		Expect(err).Should(BeNil(), "Pods are not ready after timeout")
+
+		res := kubectl.ExecPodCmd(helpers.DefaultNamespace, "xwing",
+			helpers.CurlFail("-XPOST deathstar.default.svc.cluster.local/v1/request-landing"))
```

Comment:
Move "deathstar.default.svc.cluster.local/v1/request-landing" into a string / constant.

Reply:
I keep as a string because the GSG are usually updated by people that it's not in the daily development of test/golang. with this way it's easy to grep to found the test and what is doing. If we add too much code, docs team will not understand this one. IMHO.